plugins/localstored/src/localstorage.ts

```diff
@@ -18,7 +18,20 @@ export function localstored<S, E>(options?: {
         let serializer: (s: State<S, E>) => () => string;
         let deserializer: (s: State<S, E>) => (v: string) => void;
         let stateAtRoot: State<S, E>
-        let storageEngine: StoreEngine | Storage = options?.engine || localStorage;
+        let storageEngine: StoreEngine | Storage;
+
+        // Check if the code is running in a browser environment
+        if (typeof window !== 'undefined') {
+            // Use the specified storage engine or fallback to window.localStorage
+            storageEngine = options?.engine || window.localStorage;
+        } else {
+            // Replace with your desired behavior for non-browser environments
+            storageEngine = options?.engine || {
+                getItem: (_key: string) => Promise.resolve(null),
+                setItem: (_key: string, _value: string) => Promise.resolve(),
+                removeItem: (_key: string) => Promise.resolve(),
```

Comment:
hmmm... what would it mean for the client's code? will it get null values even when it's expected State type is non nullable?

Reply:
![image](https://github.com/avkonst/hookstate/assets/45314780/01dab751-be87-4a00-a40b-4ed62e5164e5)

It's base from here

Reply:
getItem will return null
setItem will do nothing
removeItem will do nothing